src/Exercism.Representers.CSharp/Normalization/NormalizeDictionaryInitialization.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using Serilog;
+
+namespace Exercism.Representers.CSharp.Normalization
+{
+    public class NormalizeDictionaryInitialization : CSharpSyntaxRewriter
+    {
+        private class InvalidKindException : Exception
+        {
+            public SyntaxKind Kind;
+        }
+
+        public override SyntaxNode VisitInitializerExpression(InitializerExpressionSyntax node)
+        {
+            SyntaxNode DefaultVisit() => base.VisitInitializerExpression(node);
+
+            try
+            {
+                if (!IsDictionary(node))
+                {
+                    return DefaultVisit();
+                }
+
+                var initializerExtractionResults
+                    = node.Kind() switch
+                    {
+                        SyntaxKind.CollectionInitializerExpression
+                        => ExtractInitializersWithCollectionSyntax(node),
+                        SyntaxKind.ObjectInitializerExpression
+                        => ExtractInitializersWithObjectSyntax(node),
+                        _
+                        => throw new InvalidKindException {Kind = node.Kind()}
+                    };
+
+                if (!initializerExtractionResults.Success)
+                {
+                    return DefaultVisit();
+                }
+
+                var visitedInitializerSyntaxNodePairs = initializerExtractionResults.InitializerSyntaxNodePairs
+                    .Select(initializer => new KeyValuePair<SyntaxNode, SyntaxNode>(
+                        this.Visit(initializer.Key),
+                        this.Visit(initializer.Value))
+                    );
+
+                var replacementNode
+                    = BuildReplacementSyntaxWithCollectionInitialization(visitedInitializerSyntaxNodePairs);
```

Comment:
The explicit visit of the `InitializerSyntaxNodePairs` is not needed, as they will automatically be visited by the `base.VisitInitializerExpression` at the end of this method.

```suggestion
                var replacementNode
                    = BuildReplacementSyntaxWithCollectionInitialization(initializerExtractionResults.InitializerSyntaxNodePairs);
```